Source/WebKit/WebProcess/WebCoreSupport/WebNotificationClient.cpp

```diff
@@ -119,9 +123,20 @@ NotificationClient::Permission WebNotificationClient::checkPermission(ScriptExec
     callOnMainRunLoopAndWait([&resultPermission, origin = origin->data().toString().isolatedCopy()] {
         resultPermission = WebProcess::singleton().supplement<WebNotificationManager>()->policyForOrigin(origin);
     });
+
+    // To reduce fingerprinting, if the origin has not requested permission to use the
+    // Notifications API, and the permission state is "denied", return "default" instead.
+    if (!m_notificationPermissionRequesters.contains(context->securityOrigin()->data()) && resultPermission == NotificationClient::Permission::Denied)
```

Comment:
We should do the `resultPermission == NotificationClient::Permission::Denied` check first since it is cheaper:
```
if (resultPermission == NotificationClient::Permission::Denied && !m_notificationPermissionRequesters.contains(context->securityOrigin()->data()))
```

Reply:
Fixed!